### ISSUE TITLE
Authentication error after a week

### ISSUE THREAD

**Issue by Glujaz** [nicoduj/homebridge-deebotEcovacs#72]
Hi there again.

it's been a while since I came back here. The last changes you made with zones and so on works still very well !

I encounter another problem though, it's doing so since for a month now I think. Once a week, I have to restart Homebridge because an authentication error with the robot. I assume it is more related to the online service willing to disconnect and reconnect once a week.

any way to make the plugin do it automatically instead ?

thanks in advance ! :)

**Comment by Mks322** [comment]
I have the same issue, would love to see a fix to it! :)

**Comment by nicoduj** [comment]
Hi,

could you provide some logs when it happens ? I don't have such a problem here, are you using an MQTT or an xmpp deebot ?

**Comment by Glujaz** [comment]
I have to disable the automatic weekly reboot of home bridge, and I'll go back to you.
My vacuum cleaner is a deebot 905, so not working through anything like MQTT or XMP (I think ?)

**Comment by mrbungle64** [comment]
@Glujaz 

> My vacuum cleaner is a deebot 905

You mean Deebot OZMO 905, right?

**Comment by Glujaz** [comment]
Yes, i Do !

**Comment by mrbungle64** [comment]
@Glujaz 

> so not working through anything like MQTT or XMP (I think ?)

The Deebot OZMO 905 uses the MQTT protocol.

> Once a week, I have to restart Homebridge because an authentication error with the robot. I assume it is more related to the online service willing to disconnect and reconnect once a week.

Please check if it's an `RequestOAuthError: Authentication error`.
This error should not cause any problems, but please provide some logs anyway.

**Comment by Glujaz** [comment]
Ok, I finally got the error message, and it's exactly what you said :

[9/12/2021, 3:24:06 PM] [DeebotEcovacs] INFO - setDeebotEcovacsOrderCharacteristic for Hoover J. Edgar-CustomArea,start,37,24,2111,-1595,1
[9/12/2021, 3:24:06 PM] [DeebotEcovacs] INFO - Error from deebot Hoover J. Edgar(ready : true) : RequestOAuthError: Authentication error 
[9/12/2021, 3:24:06 PM] [DeebotEcovacs] INFO - updating sensor for Hoover J. Edgar : true 

So, for now, I'm using a cronjob to reboot home bridge once a week, and it's doing it's job. But didn't used this solution initially, and if there is a better one by solving this "issue", I'm all for it.

**Comment by nicoduj** [comment]
@mrbungle64 Do I need to do something on such error event ? Maybe trying to reconnect ? Right now, I am only using this event to check Timeouts, and had some error event on the motion detector, but I am not issuing anything to the lib itself.

**Comment by mrbungle64** [comment]
@nicoduj 

> @mrbungle64 Do I need to do something on such error event ? Maybe trying to reconnect ? Right now, I am only using this event to check Timeouts, and had some error event on the motion detector, but I am not issuing anything to the lib itself.

Yes, it is recommended to reconnect when receiving a "disconnect" event from the library.

**Comment by julien94800** [comment]
Hello, I have same kind of trouble. Did you find any solution for this?
Thanks in advance

**Comment by nicoduj** [comment]
Hi, could you check if it still occurs ? Can't reproduce on my side, and checking the code I disconnect / reconnect on disconnect event.

**Comment by julien94800** [comment]
Hello @nicoduj 
Hope you are doing well. I confirm that it still happened with previous version (not the one from today but the one just before). 
I will update today and let you know in some time if it still happen or not. It generally happen once a week. 
Thanks and have a nice day

**Comment by julien94800** [comment]
Hello @nicoduj 
I confirm that the problem still exist with last version. Unfortunately no idea about how I could try to debug anything. If I restart, then it works again

**Comment by julien94800** [comment]
When I get the problem I get the following messages usually ![image](https://user-images.githubusercontent.com/22552386/164914860-b72e9528-5c66-462c-abe8-91e0deb5e774.jpeg)

**Comment by julien94800** [comment]
But today I had this one too (after trying several time) ![image](https://user-images.githubusercontent.com/22552386/164914878-6601a676-be1c-42b7-b9b0-d6937e1cd217.jpeg)
